cluster/common/velero/README.md

```diff
@@ -0,0 +1,70 @@
+# Velero Terraform Module
+
+This Terraform module implements Velero restore for a cluster migration scenario. You can use this Terraform module as part of any Terraform plan. It only depends on the existence of a Kubernetes cluster.
+
+Note: This module does not implement Velero backup, contributions are welcome.
+
+Requirements:
+
+* Kubectl
+* Bash-compatible shell
+* [jq](https://stedolan.github.io/jq/)
+* Velero CLI (minimum >1.0)
+
+## Install Velero CLI
+
+There are multiple ways to install the Velero CLI. We strongly recommend that you use an [official release](https://github.com/heptio/velero/releases) of Velero. The tarballs for each release contain the velero command-line client. On OSX, you can use homebrew: `brew install velero`.
+
+There is a dependency on CLI commands that are only available in >v1.0 so please be sure to install >v1.0 or above.
+
+## Scenarios
+
+The Terraform module makes an assumption that you are using Bedrock's Terraform scripts OR have at least set `output_directory` & `kubeconfig_filename` to point to a kubeconfig file saved to disk and have set `kubeconfig_complete`.
+
+## Cluster Migration: How Restore Works
+
+The module implements restore for a cluster migration using a Terraform resource `velero_restore` that locally executes a bash script with the following process:
+
+1. Velero (server-side) is installed to the Kubernetes cluster with the provided secrets and configuration. Velero is installed in restore-only mode. If Velero is already installed in the cluster, the bash script may override it.
+2. Once Velero is installed. The bash script sleeps for at least the backup sync interval (1m) so that the Velero backup resource objects are synchronized from the storage provider. The bash script will attempt to describe the backup using the backup name provided to verify it's existence.
```

Comment:
Is there a way we could execute something in the velero cli and wait for the first sync interval versus use a fixed sync interval?  If not, that's ok, and if so, its ok to open an issue and revisit this in a future PR.

Reply:
I could continually sleep/ping the backup server using `velero backup describe` until it returns a description. I will open an issue.